storage/innobase/include/dict0mem.h

```diff
@@ -1540,7 +1540,7 @@ struct dict_table_t {
 	bool versioned() const { return vers_start || vers_end; }
 	bool versioned_by_id() const
 	{
-		return vers_start && cols[vers_start].mtype == DATA_INT;
+		return versioned() && cols[vers_start].mtype == DATA_INT;
```

Comment:
When do we have only one of `vers_start` and `vers_end`? Shouldn’t a versioned table have both columns? Why is the definition of `versioned()` using `&&` instead of `||`?

Reply:
I see this:
```
	unsigned	vers_start:10;
				/*!< System Versioning: row start col index */
	unsigned	vers_end:10;
				/*!< System Versioning: row end col index */
```
I would suggest to initialize these fields to `REC_MAX_N_FIELDS` and in `versioned()`, compare to this magic constant. In that way, the first column (column 0) will not be mistaken as "not row version start or end" even when it actually is one of the versioning columns.

Reply:
The answer is as follows:
if both vers_start and ves_end are set to 0 -- then no versioning is there.
If at least one column is nonzero, then versioning is there.

The problem was that it was somehow supposed that the versioning column can't have index 0.

I by myself don't like that bits counting, because it makes no sense for me. I can't see if somebody can optimize some cacheline fit here, by srhinking bits.
And I would prefer just adding separate field. 
But ok, let's let it be, let it check versioning by fields indexes values, but you cant determine anything by checking that only one index field is equal to zero.
The only thing is that indexes are uniqe, so vers_start can't be equal to vers_end, if those fields exist. The only special meaning is `vers_start == vers_end && vers_start == 0`

-- That's how I understand how should that design work